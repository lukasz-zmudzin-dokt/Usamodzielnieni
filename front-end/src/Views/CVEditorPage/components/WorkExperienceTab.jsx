import React from "react";
import movie_3 from "../../../assets/movie_3.png";
import CVEditorTab from 'Views/CVEditorPage/components/CvEditorTab';
import ActionWithDate from "./ActionWithDate";

const WorkExperienceTab = ({ data, onChange, onPrevClick, onNextClick, comments }) => (
    <CVEditorTab
        title="DoÅ›wiadczenie zawodowe"
        movie={movie_3}
        onPrevClick={onPrevClick}
        onNextClick={onNextClick}
        comments={comments}
    >
        <ActionWithDate data={data} onChange={onChange} />
    </CVEditorTab>
)

export default WorkExperienceTab;