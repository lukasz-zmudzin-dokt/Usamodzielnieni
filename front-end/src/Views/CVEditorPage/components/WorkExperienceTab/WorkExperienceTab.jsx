import React from "react";
import movie_3 from "assets/movie_3.png";
import { CVEditorTab, ActionWithDate } from "../";

const WorkExperienceTab = ({ data, onChange, onPrevClick, onNextClick, comments, loading, error, showComments }) => (
    <CVEditorTab
        title="DoÅ›wiadczenie zawodowe"
        movie={movie_3}
        onPrevClick={onPrevClick}
        onNextClick={onNextClick}
        comments={comments}
        loading={loading}
        error={error}
        showComments={showComments}
    >
        <ActionWithDate data={data} onChange={onChange} />
    </CVEditorTab>
)

export default WorkExperienceTab;