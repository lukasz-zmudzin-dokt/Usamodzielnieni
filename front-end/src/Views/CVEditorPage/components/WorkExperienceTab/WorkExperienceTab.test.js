import React from "react";
import { render } from "@testing-library/react";
import WorkExperienceTab from "./WorkExperienceTab";

describe("WorkExperienceTab", () => {
  let props;
  beforeEach(() => {
    props = {
      data: [],
      onChange: () => {},
    };
  });

  it("should render without crashing", () => {
    const { container } = render(<WorkExperienceTab {...props} />);

    expect(container).toMatchSnapshot();
  });
});
