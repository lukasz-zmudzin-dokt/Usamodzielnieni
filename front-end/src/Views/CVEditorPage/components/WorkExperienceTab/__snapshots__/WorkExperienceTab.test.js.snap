// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperienceTab should render without crashing 1`] = `
<div>
  <div>
    <h3>
      Doświadczenie zawodowe
    </h3>
    <div
      class="CVEditor__videoContainer"
    >
      <div
        class="fade alert alert-info show"
        role="alert"
      >
        Ładowanie...
      </div>
    </div>
    <form
      class=""
      novalidate=""
    >
      <div
        class="row"
      >
        <div
          class="form-group col-md-6 col-12"
        >
          <label
            class="form-label"
            for="startTime"
          >
            Od:
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                id="startTime"
                placeholder="Data rozpoczęcia"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="invalid-feedback"
          >
            Data rozpoczęcia jest wymagana.
          </div>
        </div>
        <div
          class="form-group col-md-6 col-12"
        >
          <label
            class="form-label"
            for="endTime"
          >
            Do:
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                id="endTime"
                placeholder="Data zakończenia"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="actionPlace"
        >
          Miejsce:
        </label>
        <input
          class="form-control"
          id="actionPlace"
          maxlength="50"
          minlength="1"
          placeholder="Nazwa szkoły/miejsca pracy"
          required=""
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        >
          Miejsce jest wymagane.
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="actionDescription"
        >
          Opis:
        </label>
        <input
          class="form-control"
          id="actionDescription"
          maxlength="400"
          minlength="1"
          placeholder="Profil/stanowisko ..."
          required=""
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        >
          Opis jest wymagany.
        </div>
      </div>
      <button
        class="mb-3 btn btn-success"
        type="submit"
      >
        + Dodaj
      </button>
    </form>
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <button
          class="form_navigation_prev btn btn-primary btn-block"
          disabled=""
          type="button"
        >
          ← Wstecz
        </button>
      </div>
      <div
        class="col-md-6 col-12"
      >
        <form
          class=""
        >
          <button
            class="btn btn-success btn-block"
            id="saveButton"
            type="submit"
          >
            Zapisz zmiany i generuj CV
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
